kratos/spatial_containers/spatial_search_result_container.cpp

```diff
@@ -0,0 +1,480 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+#include <numeric>
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "includes/data_communicator.h"
+#include "includes/node.h"
+#include "includes/geometrical_object.h"
+#include "spatial_containers/spatial_search_result_container.h"
+
+namespace Kratos
+{
+
+template <class TObjectType>
+SpatialSearchResultContainer<TObjectType>::SpatialSearchResultContainer()
+{
+    // TODO: Add something if required
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TObjectType>
+void SpatialSearchResultContainer<TObjectType>::AddResult(SpatialSearchResultType& rResult)
+{
+    // Check if the object has been found
+    if (rResult.IsObjectFound()) {
+        // Adding to the local results
+        auto p_result = Kratos::make_shared<SpatialSearchResultType>(rResult);
```

Comment:
In here you are passing a reference existing object of type ResultType (not a pointer in principle!) as input.
By default at this point we shall assume that the lifetime of such an existing object is managed elsewhere. Here you are creating a shared_pointer to it, thus effectively taking ownership of it.

The only scenario i can see in which this works is if rResult is a raw_pointer for which the lifespan **is not managed elsewhere**. HOWEVER if this is the case, then you should manage the lifetime of that object when you create it not here ...

Reply:
I think we should discuss this in detail, but check how the SpatialSearchResult class works. 

https://github.com/KratosMultiphysics/Kratos/blob/ba68022e88f7df806271798fa81dbd8eed3d4142/kratos/spatial_containers/spatial_search_result.h#L50

SpatialSearchResult works with GlobalPointer and in the constructor considers a raw pointer: https://github.com/KratosMultiphysics/Kratos/blob/ba68022e88f7df806271798fa81dbd8eed3d4142/kratos/spatial_containers/spatial_search_result.h#L67

Reply:
I will see if I can avoid that shared_ptr now that you stated that you are not meaning to enforce intrussive_ptr.

Reply:
I tried with unique_ptr but gave a lot of issues